pkg/controller/tiers/tiers_controller.go

```diff
@@ -188,20 +192,37 @@ func (r *ReconcileTiers) prepareTiersConfig(ctx context.Context, reqLogger logr.
 				}
 				return nil, &reconcile.Result{RequeueAfter: 10 * time.Second}
 			}
-			kubeDNSIPs := kubeDNSService.Spec.ClusterIPs
-
-			for _, IP := range kubeDNSIPs {
-				var builder strings.Builder
-				builder.WriteString(IP)
-				if net.ParseIP(IP).To4() != nil {
-					builder.WriteString("/32")
-					tiersConfig.DNSEgressCIDRs.IPV4 = append(tiersConfig.DNSEgressCIDRs.IPV4, builder.String())
+
+			var kubeDNSIPs []string
+			for _, service := range kubeDNSServiceList.Items {
+				if strings.Contains(service.Name, "upstream") {
+					continue
 				} else {
-					builder.WriteString("/128")
-					tiersConfig.DNSEgressCIDRs.IPV6 = append(tiersConfig.DNSEgressCIDRs.IPV6, builder.String())
+					kubeDNSIPs = service.Spec.ClusterIPs
 				}
+			}
 
+			if len(kubeDNSIPs) > 0 {
+				for _, IP := range kubeDNSIPs {
+					var builder strings.Builder
+					builder.WriteString(IP)
+					if net.ParseIP(IP).To4() != nil {
+						builder.WriteString("/32")
+						tiersConfig.DNSEgressCIDRs.IPV4 = append(tiersConfig.DNSEgressCIDRs.IPV4, builder.String())
+					} else {
+						builder.WriteString("/128")
+						tiersConfig.DNSEgressCIDRs.IPV6 = append(tiersConfig.DNSEgressCIDRs.IPV6, builder.String())
+					}
+
+				}
+			} else {
+				r.status.SetDegraded(operatorv1.ResourceReadError,
```

Comment:
We may want to adjust the wording for the two `SetDegraded` reasons to distinguish them clearly from each other